consume.go

```diff
@@ -291,6 +307,68 @@ func (cmd *consumeCmd) newConsumedMessage(m *sarama.ConsumerMessage) (consumedMe
 	return result, nil
 }
 
+func (cmd *consumeCmd) encoderForType(typ string) (func([]byte) (json.RawMessage, error), error) {
+	var enc func([]byte) string
+	switch typ {
+	case "json":
+		return func(data []byte) (json.RawMessage, error) {
```

Comment:
At this point I'd probably consider pushing the generation of this function (and the one like it below) out to named functions of their own, just because the switch statement is getting a bit long.

Reply:
True. I will do it in another PR.

Reply:
I tried to "simplify" the switch using a map with dedicated method per type:  https://github.com/heetch/hkt/commit/0e4a242642b9919fa9e72ea613c4435b5000a163 If you think it's okay, I can propose the PR (it's based on this branch).

Reply:
Personally I don't think it's too unwieldy currently - not sure that splitting it up will help readability.